### PR TITLE
Star sign

### PR DESCRIPTION
**Pull request by zedlang**

### CODE REVIEW

File: app/models/profile.rb [hunk 1]
```diff
@@ -1,3 +1,36 @@
...
+                  gemini: [Date.new(year,5,21),Date.new(year,6,20)],
+                  cancer: [Date.new(year,6,21),Date.new(year,7,22)],
+                  leo: [Date.new(year,7,23),Date.new(year,8,22)],
+                  virgo: [Date.new(year,8,23),Date.new(year,9,22)],
+                  libra: [Date.new(year,9,23), Date.new(year,10,22)],
+                  scorpio: [Date.new(year,10,23),Date.new(year,11,21)],
+                  sagittarius: [Date.new(year,11,22),Date.new(year,12,21)]
+                  }
+
+    self.star_sign = star_signs.select do |key,value|
```
Comment (dmgarland) [inline app/models/profile.rb:37]: If you only want one result, use `find` instead of `select`

File: features/step_definitions/star_sign_steps.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    confirmed_at: Time.now,
+    date_of_birth: Date.new(1966,2,12))
+
+  @profile = Profile.create!(
+    biography: "A long time ago, in a galaxy far away...", 
+    customer: @customer)
+end
+
+Then(/^their star sign is automatically set$/) do
+  star_signs = ["Capricorn","Aquarius","Pisces","Aries","Taurus","Gemini",
```
Comment (dmgarland) [inline features/step_definitions/star_sign_steps.rb:16]: Maybe we could put this in a constant on a module somewhere and share this list between your test and your model?

File: app/models/profile.rb [hunk 3]
```diff
@@ -1,3 +1,36 @@
...
+
+  attr_accessor :star_sign
+
+  private
+
+  def set_star_sign
+
+    dob = self.customer.date_of_birth
+    year = dob.year
+    month = dob.month
```
Comment (dmgarland) [inline app/models/profile.rb:20]: where are month and day used?

File: features/step_definitions/star_sign_steps.rb (hunk 2 above)
Comment (zedlang) [inline features/step_definitions/star_sign_steps.rb:16]: My thought is that this is a fairly minor bit of duplication that I can't see being used elsewhere

File: app/models/profile.rb [hunk 4]
```diff
@@ -7,4 +7,36 @@ def display_attribute_keys
...
+    day = dob.day
+
+    star_signs = {capricorn: [Date.new(year,12,20),Date.new(year, 1,19)],
```
Comment (dmgarland) [inline app/models/profile.rb:23]: I'll merge this but I think this data ought to be moved somewhere else.

**Pull request merged**